ruma-common/src/push/condition.rs

```diff
@@ -151,4 +348,126 @@ mod tests {
             } if key == "room"
         );
     }
+
+    #[test]
+    fn words_match() {
+        assert!(matches_word("foo bar", "foo"));
+        assert!(!matches_word("Foo bar", "foo"));
```

Comment:
Why should this match fail?

Reply:
Because `matches_word` is case sensitive. Case insensitivity is part of `matches_pattern` (that in turn calls `matches_word`. This test is unnecessary for our use case though.

Reply:
Why though? Is it specced that some of these push conditions are case sensitive and others case insensitive? Is that Synapse behavior we're copying?

Reply:
It is not specced, I'm copying the behavior of Synapse. But this issue implies that case is insensitive by default: https://github.com/matrix-org/matrix-doc/issues/714

Reply:
Okay, but we only end up calling `matches_word` through `matches_pattern` and so in the end all matches really are case-insensitive? In that case I'm okay with it.

Reply:
Yes, that's what happens.